workbench/test/selenium_test.py

```diff
@@ -19,3 +23,15 @@ def setUp(self):
         # Clear the in-memory key value store, the usage store, and whatever
         # else needs to be cleared and re-initialized.
         reset_global_state()
+
+    @contextmanager
+    def wait_for_page_load(self, timeout=30):
+        """
+        Uses Selenium's built-in "staleness" hook to wait until the page has
+        loaded. For use when clicking a link to ensure that the new page
+        has loaded before selecting elements.
+        """
+        old_page = self.browser.find_element_by_tag_name('html')
+        yield WebDriverWait(self.browser, timeout).until(
+            staleness_of(old_page)
+        )
```

Comment:
You're right- I pulled this from http://www.obeythetestinggoat.com/how-to-get-selenium-to-wait-for-page-load-after-a-click.html but I think their example is slightly wrong and I should have wrapped `link.click()` in the context manager instead.